app/src/main/java/nodomain/freeyourgadget/gadgetbridge/service/devices/no1f1/No1F1Support.java

```diff
@@ -430,4 +445,48 @@ private void stopNotification() {
         } catch (IOException e) {
         }
     }
+
+    private void handleStepData(byte[] data) {
+        if (data[1] == (byte) 0xfd) {
+            // TODO Check CRC
+            if (samples.size() > 0) {
+                try (DBHandler dbHandler = GBApplication.acquireDB()) {
+                    Long userId = DBHelper.getUser(dbHandler.getDaoSession()).getId();
+                    Long deviceId = DBHelper.getDevice(getDevice(), dbHandler.getDaoSession()).getId();
+                    No1F1SampleProvider provider = new No1F1SampleProvider(getDevice(), dbHandler.getDaoSession());
+                    for (int i = 0; i < samples.size(); i++) {
+                        samples.get(i).setDeviceId(deviceId);
+                        samples.get(i).setUserId(userId);
+                        samples.get(i).setRawKind(ActivityKind.TYPE_ACTIVITY);
+                        provider.addGBActivitySample(samples.get(i));
+                    }
+                    samples.clear();
+                    LOG.info("Steps data saved");
+                    if (getDevice().isBusy()) {
+                        getDevice().unsetBusyTask();
+                        getDevice().sendDeviceUpdateIntent(getContext());
+                    }
+                } catch (Exception ex) {
```

Comment:
Please, don't ever swallow exceptions. At the very least, log them with 
`LOG.warn("message", ex);`
But also consider showing a toast -- I guess the user would be thankful to know that his activity data went down the hill instead of being saved in the database :-)

Reply:
Fixed (and for other exceptions as well).